sycl/plugins/esimd_cpu/pi_esimd_cpu.cpp

```diff
@@ -637,6 +375,32 @@ template <class KernelType, class KernelArgType, int DIMS> class libCMBatch {
   }
 };
 
+void sycl_get_cm_buffer_params(void *PtrInput, char **BaseAddr, uint32_t *Width,
+                               std::mutex *MtxLock) {
+  _pi_buffer *buf = static_cast<_pi_buffer *>(PtrInput);
+
+  *BaseAddr = cm_support::get_surface_base_addr(buf->SurfaceIndex);
+  *Width = static_cast<uint32_t>(buf->Size);
+
+  // '=' assignment causes build failure for std::mutex
+  std::memcpy(MtxLock, &(buf->mutexLock), sizeof(std::mutex));
+}
+
+void sycl_get_cm_image_params(void *PtrInput, char **BaseAddr, uint32_t *Width,
+                              uint32_t *Height, uint32_t *Bpp,
+                              std::mutex *MtxLock) {
+  _pi_image *img = static_cast<_pi_image *>(PtrInput);
+
+  *BaseAddr = cm_support::get_surface_base_addr(img->SurfaceIndex);
+
+  *Bpp = static_cast<uint32_t>(img->BytesPerPixel);
+  *Width = static_cast<uint32_t>(img->Width) * (*Bpp);
+  *Height = static_cast<uint32_t>(img->Height);
+
+  // '=' assignment causes build failure for std::mutex
+  std::memcpy(MtxLock, &(img->mutexLock), sizeof(std::mutex));
```

Comment:
why do you do a copy of mutex (generally unsafe) rather than returning the pointer to the existing one?

Reply:
Do you mean return the mutex value directly? I tried that but it fails with same build error as '=' assignment.

`mutex(const mutex*&) = delete;` from `std_mutex.h`

Reply:
not the value (since that needs copying) but a pointer (or reference) to the mutex.

Reply:
Oh, I see. I'll apply suggestion.

Reply:
Suggestion applied.